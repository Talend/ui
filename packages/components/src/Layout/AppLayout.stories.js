import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Drawer from '../Drawer';
import HeaderBar from '../HeaderBar';
import IconsProvider from '../IconsProvider';
import Layout from '.';
import SidePanel from '../SidePanel';
import SubHeaderBar from '../SubHeaderBar';
import List from '../List/ListComposition';

const actions = [
	{
		label: 'Preparations',
		icon: 'talend-dataprep',
		onClick: action('Preparations clicked'),
	},
	{
		label: 'Datasets',
		icon: 'talend-folder',
		onClick: action('Datasets clicked'),
	},
	{
		label: 'Favorites',
		icon: 'talend-star',
		onClick: action('Favorites clicked'),
	},
];

const drawers = [
	<Drawer stacked title="Hello drawers" footerActions={{}}>
		<p>You should not being able to read this because I&apos;m first</p>
	</Drawer>,
	<Drawer title="Hello drawers" footerActions={{}}>
		<p>The content dictate the width</p>
	</Drawer>,
];

const content = (
	<div>
		<h1>Welcome to the content for testing scroll</h1>
		<ul>
			{[...new Array(138)].map((_, index) => (
				<li key={index}>one</li>
			))}
		</ul>
	</div>
);
const sidePanel = <SidePanel actions={actions} />;
const dockedSidePanel = <SidePanel actions={actions} docked />;
const header = <HeaderBar brand={{ label: 'Example App Name' }} />;
const subHeader = <SubHeaderBar title="MyTitle" onGoBack={action('SubHeader onGoBack')} />;
const footer = 'Footer content';

const items = [...new Array(61)].map((_, index) => ({
	id: index,
	name: 'Title',
	created: '2016-09-22',
	modified: '2016-09-22',
	author: 'Jean-Pierre DUPONT',
	display: 'text',
}));

const tabs = {
	id: 'tab-bar',
	items: [
		{
			key: '1',
			label: 'Tab1',
		},
		{
			key: '2',
			label: 'Tab2',
		},
		{
			key: '3',
			label: 'Tab3',
		},
	],
	onSelect: action('onSelect'),
	selectedKey: '2',
};

const stories = storiesOf('Layout/AppLayout', module).addDecorator(story => (
	<div>
		<IconsProvider />
		{story()}
	</div>
));

const appStyle = require('../../stories/config/themes.scss');

/**
 * Generate story for <Layout/> component
 *
 * @param layoutStoryName Story name to display in storybook
 * @param layoutStoryProps Props to pass to <Layout/> component
 * @param layoutStoryContent Optional custom children
 */
function layoutStory(layoutStoryName, layoutStoryProps, layoutStoryContent = content) {
	stories.add(layoutStoryName, () => <Layout {...layoutStoryProps}>{layoutStoryContent}</Layout>);
}

/**
 * Generate story and its variations for <Layout/> component
 *
 * @param layoutStoryName Story name to display in storybook
 * @param layoutStoryProps Props to pass to <Layout/> component
 * @param layoutStoryContent Optional custom children
 */
function decoratedLayoutStory(layoutStoryName, layoutStoryProps, layoutStoryContent = content) {
	layoutStory(layoutStoryName, layoutStoryProps, layoutStoryContent);
	Layout.TALEND_T7_THEME_APPS.forEach(app => {
		const decoratedPropsWithTheme = {
			...layoutStoryProps,
			// hasTheme: true, should be enabled if we have one and only one Layout theme scss import
		};
		stories.add(`ðŸŽ¨ [${app.toUpperCase()}] ${layoutStoryName} `, () => (
			<div className={appStyle[app]}>
				<div className={Layout.TALEND_T7_THEME_CLASSNAME}>
					<Layout {...decoratedPropsWithTheme}>{layoutStoryContent}</Layout>
				</div>
			</div>
		));
	});
}

decoratedLayoutStory('OneColumn', {
	header,
	mode: 'OneColumn',
});

decoratedLayoutStory('OneColumn with tabs', {
	header,
	tabs,
	mode: 'OneColumn',
});

decoratedLayoutStory('TwoColumns', {
	header,
	mode: 'TwoColumns',
	one: sidePanel,
});

decoratedLayoutStory('TwoColumns with tabs', {
	header,
	tabs,
	mode: 'TwoColumns',
	one: sidePanel,
});

decoratedLayoutStory(
	'TwoColumns with big Table list',
	{
		header,
		mode: 'TwoColumns',
		one: sidePanel,
	},
	<List.Manager id="my-list" collection={items}>
		<List.Toolbar>
			<List.DisplayMode id="my-list-displayMode" />
		</List.Toolbar>
		<List.VList id="my-vlist">
			<List.VList.Text label="Id" dataKey="id" />
			<List.VList.Text label="Name" dataKey="name" />
			<List.VList.Text label="Author" dataKey="author" />
			<List.VList.Text label="Created" dataKey="created" />
			<List.VList.Text label="Modified" dataKey="modified" />
		</List.VList>
	</List.Manager>,
);

decoratedLayoutStory(
	'TwoColumns with big Large list',
	{
		header,
		mode: 'TwoColumns',
		one: sidePanel,
	},
	<List.Manager id="my-list" collection={items}>
		<List.Toolbar>
			<List.DisplayMode id="my-list-displayMode" initialValue="large" />
		</List.Toolbar>
		<List.VList id="my-vlist">
			<List.VList.Text label="Id" dataKey="id" />
			<List.VList.Text label="Name" dataKey="name" />
			<List.VList.Text label="Author" dataKey="author" />
			<List.VList.Text label="Created" dataKey="created" />
			<List.VList.Text label="Modified" dataKey="modified" />
		</List.VList>
	</List.Manager>,
);

decoratedLayoutStory('TwoColumns docked', {
	header,
	mode: 'TwoColumns',
	one: dockedSidePanel,
});

decoratedLayoutStory('TwoColumns with drawers', {
	header,
	mode: 'TwoColumns',
	one: sidePanel,
	drawers,
});

decoratedLayoutStory('OneColumn with footer', {
	header,
	mode: 'OneColumn',
	footer,
});

layoutStory('OneColumn without header', {
	mode: 'OneColumn',
});

decoratedLayoutStory('OneColumn with subHeader', {
	header,
	subHeader,
	mode: 'OneColumn',
	footer,
});

decoratedLayoutStory('TwoColumns with subHeader', {
	header,
	subHeader,
	one: sidePanel,
	mode: 'TwoColumns',
	footer,
});

layoutStory('Only subHeader', {
	subHeader,
});
