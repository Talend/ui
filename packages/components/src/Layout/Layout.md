# Layout component

This component provide two common layouts to build your page

The structure is:

* header (optional)
* content (one or two coloums)
* footer (optional)

Props table:

This component use [Inject](../Inject/Inject.md).

| name | type | description |
| -- | -- | -- |
| mode | `string` | `OneColumn` (default) or `TwoColums` |
| header | `string|object|Array|react element` | display the header |
| subheader | `string|object|Array|react element` | displayed after the header |
| drawers | `Array of <string|object|Array|react element>` | displayed in stacked drawers |
| one | `string|object|Array|react element` | displayed first colums in the TwoColumns layout |
| content | `string|object|Array|react element` | displayed the main content |
| footer | `string|object||Array|react element` | display the footer |
| getComponent | `function`| optional function to support non react element in header, subheader, ...|
