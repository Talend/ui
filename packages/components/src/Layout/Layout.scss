////
/// Layout
/// @group Custom widgets
////

/// Layout height of the header
/// @type Number (Unit)
$tc-drawer-z-index: $drawer-z-index !default;
$tc-layout-header-height: $navbar-height !default;
$tc-layout-skip-links-z-index: $tc-drawer-z-index + 20 !default;
$tc-layout-header-z-index: $tc-drawer-z-index + 10 !default;

.layout {
	display: flex;
	flex-direction: column;
	height: 100vh;
	width: 100vw;

	.skip-links {
		z-index: $tc-layout-skip-links-z-index;
	}

	.header {
		flex: 0 0 $tc-layout-header-height;
		z-index: $tc-layout-header-z-index;
	}

	.footer {
		flex-shrink: 0;
		flex-grow: 0;
	}
}
