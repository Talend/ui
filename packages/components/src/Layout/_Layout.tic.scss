.t7,
:global(.t7) {
	$brand-primary-t7: #009597;
	$brand-secondary-t7: #168396;
	@include layout-colors($brand-primary-t7, $brand-secondary-t7);

	[role='navigation']:before {
		background-position: -109.5px bottom;
		background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIzMTAiIGhlaWdodD0iMzEwIiB2aWV3Qm94PSIwIDAgMzEwIDMxMCI+ICA8cGF0aCBmaWxsPSIjMTY4Mzk2IiBkPSJNMTcxLjgxNjUyNiwyODMuMTQ1MjkyIEMxNjkuNTM3NDg1LDI3OC4zMTY4MTUgMTY1LjIxMTE3LDI3NC42NDcxNzMgMTU5Ljk3NzEwMSwyNzMuMjM3MjU4IEwxNTkuOTc3MTAxLDIwMi42MDYyOTYgTDIwNi41MDQzMDUsMjAyLjY0NDkyNCBDMjI3LjEzMTU1OSwyMDIuNjQ0OTI0IDI0My44NTc0MDMsMTg1LjQ3NDg1OSAyNDMuODU3NDAzLDE2NC4yODc1MDIgQzI0My44NTc0MDMsMTQzLjEwMDE0NSAyMjcuMTMxNTU5LDEyNS45MTA3NjcgMjA2LjUwNDMwNSwxMjUuOTEwNzY3IEMyMDQuNzY2MDU0LDEyNS45MTA3NjcgMjAzLjAyNzgwMiwxMjYuMDg0NTkyIDIwMS4zNDc0OTIsMTI2LjMzNTY3MyBDMTk2LjY3MzUyNiwxMDQuNzQyNzIzIDE3Ny40NzU1MDEsODguNTU3NjY4NSAxNTQuNDcyNjM3LDg4LjU1NzY2ODUgQzEzNi4xNjMwNTIsODguNTU3NjY4NSAxMjAuMjQ4MzkyLDk4LjgxMzM1MzYgMTEyLjE1NTg2NCwxMTMuOTE2ODMgQzExMC44MDM4OTEsMTEzLjY2NTc0OSAxMDkuNDMyNjAzLDExMy40NTMyOTYgMTA4LjAwMzM3NCwxMTMuNDUzMjk2IEM5Ni4wMDk0MzcxLDExMy40NTMyOTYgODYuMjk0NTQxMywxMjMuNDE5MjczIDg2LjI5NDU0MTMsMTM1Ljc0MTU0NiBDODYuMjk0NTQxMywxMzguNzczODI5IDg2LjkxMjU4NjQsMTQxLjcwOTU0NCA4Ny45OTQxNjUyLDE0NC4zMzYyMzUgQzc0LjY4Njg4MjYsMTQ2Ljg0NzA0MyA2NC41NjYzOTQ3LDE1OC43NjM3MjQgNjQuNTY2Mzk0NywxNzMuMTUyNTg2IEM2NC41NjYzOTQ3LDE4OS4zOTU1ODMgNzcuMzcxNTE1OCwyMDIuNTI5MDQgOTMuMTcwMjkyNiwyMDIuNTI5MDQgTDE1MC41MzI2LDIwMi41ODY5ODIgTDE1MC41MzI2LDI3My4xOTg2MyBDMTQ1LjIwMTk2MSwyNzQuNTg5MjMxIDE0MC44MzcwMTgsMjc4LjI3ODE4OCAxMzguNTE5MzQ5LDI4My4xODM5MiBDODAuNDIzMTEzMywyNzUuNzI4NzUyIDM0LjMwMTUwMDgsMjI5LjU0OTE5NyAyNi45ODE1Mjk2LDE3MS4zOTUwMiBDMzIuODMzNjQzNywxNjguMzI0MTA5IDM2Ljg1MDkzNjYsMTYyLjE2Mjk3MiAzNi44NTA5MzY2LDE1NS4wOTQwODIgQzM2Ljg1MDkzNjYsMTQ4LjAyNTE5MSAzMi44NTI5NTc3LDE0MS45MDI2ODMgMjcuMDAwODQzNSwxMzguNzkzMTQzIEMzNC4zOTgwNzAzLDgwLjY1ODI4IDgwLjYzNTU2NjMsMzQuNDc4NzI1OCAxMzguODA5MDU3LDI3LjEzOTQ0MDcgQzE0MS44NDEzNDEsMzMuMTY1MzgwMSAxNDguMDYwNDE5LDM3LjI3OTI0MjUgMTU1LjI0NTE5MywzNy4yNzkyNDI1IEMxNjIuNDEwNjUzLDM3LjI3OTI0MjUgMTY4LjYxMDQxOCwzMy4xODQ2OTQgMTcxLjY2MjAxNSwyNy4xOTczODI1IEMyMjkuNDY4NTQyLDM0LjY5MTE3ODggMjc1LjM5NzAxNSw4MC41NDIzOTY2IDI4Mi45NjgwNjcsMTM4LjM0ODkyMyBDMjc2LjU5NDQ3OCwxNDEuMjQ2MDEgMjcyLjE3MTU5MywxNDcuNjM4OTEzIDI3Mi4xNzE1OTMsMTU1LjA5NDA4MiBDMjcyLjE3MTU5MywxNjIuNTQ5MjUgMjc2LjYxMzc5MiwxNjguOTYxNDY4IDI4Mi45ODczODEsMTcxLjg1ODU1NCBDMjc1LjQ5MzU4NSwyMjkuNjY1MDgxIDIyOS42MjMwNTMsMjc1LjU3NDI0IDE3MS44MTY1MjYsMjgzLjE0NTI5MiBNMzA5LjAyMjUyOSwxNTUuMDk0MDgyIEMzMDkuMDIyNTI5LDE0Ni4zMjU1NjggMzAyLjkwMDAyLDEzOS4wMDU1OTYgMjk0LjcxMDkyMywxMzcuMTMyMTQ3IEMyODYuNjE4Mzk2LDczLjg3OTA5ODMgMjM2LjU5NTM3NCwyMy43NDAxOTI5IDE3My4zNjE2MzksMTUuNTMxNzgyIEMxNzEuNzc3ODk4LDYuOTM3MDkyODkgMTY0LjI4NDEwMiwwLjQyODMwNTg2OCAxNTUuMjQ1MTkzLDAuNDI4MzA1ODY4IEMxNDYuMjI1NTk4LDAuNDI4MzA1ODY4IDEzOC43NTExMTYsNi45MTc3Nzg5OCAxMzcuMTQ4MDYxLDE1LjQ3Mzg0MDMgQzczLjc1OTgxNSwyMy41MDg0MjYgMjMuNTQzNjU0LDczLjYwODcwMzYgMTUuMzM1MjQzLDEzNi45NTgzMjIgQzYuNjQzOTg0MzgsMTM4LjQyNjE3OSAwLDE0NS45NTg2MDMgMCwxNTUuMDk0MDgyIEMwLDE2NC4yMTAyNDYgNi42MjQ2NzA0NywxNzEuNzQyNjcxIDE1LjMxNTkyOTEsMTczLjI0OTE1NSBDMjMuNDQ3MDg0NCwyMzYuNjc2MDMgNzMuNzIxMTg3MiwyODYuODcyODc3IDEzNy4xODY2ODksMjk0Ljg4ODE0OSBDMTM4Ljk2MzU2OSwzMDMuMTkzMTI5IDE0Ni4zNDE0ODIsMzA5LjQzMTUyMSAxNTUuMTg3MjUxLDMwOS40MzE1MjEgQzE2NC4wMzMwMjEsMzA5LjQzMTUyMSAxNzEuNDEwOTM0LDMwMy4xNzM4MTUgMTczLjE4NzgxNCwyOTQuODQ5NTIxIEMyMzYuNTU2NzQ2LDI4Ni42OTkwNTEgMjg2LjY5NTY1MiwyMzYuNDQ0MjYzIDI5NC43MzAyMzcsMTczLjAzNjcwMiBDMzAyLjkxOTMzNCwxNzEuMTQzOTM5IDMwOS4wMjI1MjksMTYzLjg0MzI4MiAzMDkuMDIyNTI5LDE1NS4wOTQwODIiLz48L3N2Zz4=');
	}
}
