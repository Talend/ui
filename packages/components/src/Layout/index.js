import Layout from './Layout.component';

export default Layout;
