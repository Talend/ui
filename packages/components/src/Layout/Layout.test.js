import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { SubHeaderBar } from '../SubHeaderBar/SubHeaderBar.component';

import Layout from './Layout.component';

jest.mock('react-dom');

const header = <h1>Header</h1>;
const subHeader = <SubHeaderBar title="defaultTitle" onGoBack={jest.fn()} />;
const footer = <h3>Footer</h3>;
const one = <h1>Column one</h1>;
const two = <h1>Column two</h1>;
const tabs = {
	items: [
		{
			id: 'tab-bar-action-1',
			key: '1',
			label: 'Tab1',
		},
		{
			id: 'tab-bar-action-2',
			key: '2',
			label: 'Tab2',
		},
		{
			id: 'tab-bar-action-3',
			key: '3',
			label: 'Tab3',
		},
	],
	onSelect: jest.fn(),
	selectedKey: '2',
};
const drawers = [
	<div style={{ width: 500 }}>
		<h1>Hello drawers</h1>
		<p>You should not being able to read this because I&#39;m first</p>
	</div>,
	<div style={{ width: 400 }}>
		<h1>Hello drawers</h1>
		<p>The content dictate the width</p>
	</div>,
];

describe('Layout', () => {
	it('should render Layout OneColumn', () => {
		const wrapper = renderer
			.create(
				<Layout mode="OneColumn" header={header}>
					{one}
				</Layout>,
			)
			.toJSON();

		expect(wrapper).toMatchSnapshot();
	});

	it('should render Layout TwoColumns', () => {
		const wrapper = renderer
			.create(
				<Layout mode="TwoColumns" one={one} header={header}>
					{two}
				</Layout>,
			)
			.toJSON();

		expect(wrapper).toMatchSnapshot();
	});

	it('should render layout with Drawer component', () => {
		const wrapper = renderer
			.create(
				<Layout mode="TwoColumns" one={one} header={header} drawers={drawers}>
					{two}
				</Layout>,
			)
			.toJSON();

		expect(wrapper).toMatchSnapshot();
	});

	it('should render layout with footer component', () => {
		const wrapper = renderer
			.create(
				<Layout mode="OneColumn" header={header} footer={footer}>
					{one}
				</Layout>,
			)
			.toJSON();

		expect(wrapper).toMatchSnapshot();
	});

	it('should render layout without header', () => {
		const wrapper = renderer.create(<Layout mode="OneColumn">{one}</Layout>).toJSON();

		expect(wrapper).toMatchSnapshot();
	});

	it('should render layout with TabBar component', () => {
		const wrapper = renderer
			.create(
				<Layout mode="TwoColumns" one={one} header={header} tabs={tabs}>
					{two}
				</Layout>,
			)
			.toJSON();

		expect(wrapper).toMatchSnapshot();
	});
	it('should render layout with SubHeader', () => {
		const wrapper = shallow(subHeader);
		expect(wrapper).toMatchSnapshot();
	});
	it('should render layout with subHeader in OneColumn mode', () => {
		const wrapper = shallow(
			<Layout subHeader={subHeader} mode="OneColumn">
				{one}
			</Layout>,
		);
		expect(wrapper.instance().props.subHeader).toEqual(subHeader);
		expect(wrapper.instance().props.mode).toEqual('OneColumn');
	});
	it('should render layout with subHeader in TwoColumns mode', () => {
		const wrapper = shallow(
			<Layout subHeader={subHeader} mode="TwoColumns">
				{one}
			</Layout>,
		);
		expect(wrapper.instance().props.subHeader).toEqual(subHeader);
		expect(wrapper.instance().props.mode).toEqual('TwoColumns');
	});
});
