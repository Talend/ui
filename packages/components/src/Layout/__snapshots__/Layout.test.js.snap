// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render Layout OneColumn 1`] = `
<div
  className="tc-layout theme-layout"
  id={undefined}
>
  <header
    className="tc-layout-header theme-header"
    role="banner"
  >
    <h1>
      Header
    </h1>
  </header>
  <div
    className="tc-layout-one-column theme-main"
  >
    <div
      className="theme-tc-with-drawer"
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "overflow": "auto",
          }
        }
      >
        <h1>
          Column one
        </h1>
      </div>
      <div
        className="theme-tc-with-drawer-container"
      />
    </div>
  </div>
</div>
`;

exports[`Layout should render Layout TwoColumns 1`] = `
<div
  className="tc-layout theme-layout"
  id={undefined}
>
  <header
    className="tc-layout-header theme-header"
    role="banner"
  >
    <h1>
      Header
    </h1>
  </header>
  <div
    className="tc-layout-two-columns theme-container"
  >
    <div
      className="tc-layout-two-columns-left theme-sidemenu"
    >
      <h1>
        Column one
      </h1>
    </div>
    <div
      className="tc-layout-two-columns-main theme-main"
    >
      <div
        className="theme-tc-with-drawer"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "height": "100%",
              "overflow": "auto",
            }
          }
        >
          <h1>
            Column two
          </h1>
        </div>
        <div
          className="theme-tc-with-drawer-container"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout should render layout with Drawer component 1`] = `
<div
  className="tc-layout theme-layout"
  id={undefined}
>
  <header
    className="tc-layout-header theme-header"
    role="banner"
  >
    <h1>
      Header
    </h1>
  </header>
  <div
    className="tc-layout-two-columns theme-container"
  >
    <div
      className="tc-layout-two-columns-left theme-sidemenu"
    >
      <h1>
        Column one
      </h1>
    </div>
    <div
      className="tc-layout-two-columns-main theme-main"
    >
      <div
        className="theme-tc-with-drawer"
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "height": "100%",
              "overflow": "auto",
            }
          }
        >
          <h1>
            Column two
          </h1>
        </div>
        <div
          className="theme-tc-with-drawer-container"
        >
          <div
            className="tc-with-drawer-wrapper"
            onTransitionEnd={[Function]}
            style={
              Object {
                "transform": "translateX(100%)",
              }
            }
          >
            <span
              style={
                Object {
                  "opacity": 0.9,
                }
              }
            />
            <div
              style={
                Object {
                  "width": 500,
                }
              }
              transitioned={false}
            >
              <h1>
                Hello drawers
              </h1>
              <p>
                You should not being able to read this because I'm first
              </p>
            </div>
          </div>
          <div
            className="tc-with-drawer-wrapper"
            onTransitionEnd={[Function]}
            style={
              Object {
                "transform": "translateX(100%)",
              }
            }
          >
            <span
              style={
                Object {
                  "opacity": 0.9,
                }
              }
            />
            <div
              style={
                Object {
                  "width": 400,
                }
              }
              transitioned={false}
            >
              <h1>
                Hello drawers
              </h1>
              <p>
                The content dictate the width
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout should render layout with SubHeader 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubHeaderBar
    onGoBack={[Function]}
    t={[Function]}
    title="defaultTitle"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <ActionBar
          className="theme-tc-subheader-navbar tc-subheader-navbar"
>
          <SubHeaderBarActions
                    left={true}
          >
                    <Action
                              bsStyle="link"
                              className="theme-tc-subheader-back-button tc-subheader-back-button"
                              hideLabel={true}
                              icon="talend-arrow-left"
                              id="backArrow"
                              label="Go Back"
                              onClick={[Function]}
                    />
                    <InputTitleSubHeader
                              editMode={false}
                              inProgress={false}
                              loading={false}
                              t={[Function]}
                              title="defaultTitle"
                    />
          </SubHeaderBarActions>
</ActionBar>,
        null,
      ],
      "className": "theme-tc-subheader tc-subheader",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            null,
            <SubHeaderBarActions
              left={true}
>
              <Action
                            bsStyle="link"
                            className="theme-tc-subheader-back-button tc-subheader-back-button"
                            hideLabel={true}
                            icon="talend-arrow-left"
                            id="backArrow"
                            label="Go Back"
                            onClick={[Function]}
              />
              <InputTitleSubHeader
                            editMode={false}
                            inProgress={false}
                            loading={false}
                            t={[Function]}
                            title="defaultTitle"
              />
</SubHeaderBarActions>,
            false,
            null,
            false,
            null,
            false,
            null,
          ],
          "className": "theme-tc-subheader-navbar tc-subheader-navbar",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                null,
                <Action
                  bsStyle="link"
                  className="theme-tc-subheader-back-button tc-subheader-back-button"
                  hideLabel={true}
                  icon="talend-arrow-left"
                  id="backArrow"
                  label="Go Back"
                  onClick={[Function]}
/>,
                null,
                <InputTitleSubHeader
                  editMode={false}
                  inProgress={false}
                  loading={false}
                  t={[Function]}
                  title="defaultTitle"
/>,
                null,
              ],
              "left": true,
            },
            "ref": null,
            "rendered": Array [
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "bsStyle": "link",
                  "className": "theme-tc-subheader-back-button tc-subheader-back-button",
                  "hideLabel": true,
                  "icon": "talend-arrow-left",
                  "id": "backArrow",
                  "label": "Go Back",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              null,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "editMode": false,
                  "inProgress": false,
                  "loading": false,
                  "t": [Function],
                  "title": "defaultTitle",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              null,
            ],
            "type": [Function],
          },
          false,
          null,
          false,
          null,
          false,
          null,
        ],
        "type": [Function],
      },
      null,
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <ActionBar
            className="theme-tc-subheader-navbar tc-subheader-navbar"
>
            <SubHeaderBarActions
                        left={true}
            >
                        <Action
                                    bsStyle="link"
                                    className="theme-tc-subheader-back-button tc-subheader-back-button"
                                    hideLabel={true}
                                    icon="talend-arrow-left"
                                    id="backArrow"
                                    label="Go Back"
                                    onClick={[Function]}
                        />
                        <InputTitleSubHeader
                                    editMode={false}
                                    inProgress={false}
                                    loading={false}
                                    t={[Function]}
                                    title="defaultTitle"
                        />
            </SubHeaderBarActions>
</ActionBar>,
          null,
        ],
        "className": "theme-tc-subheader tc-subheader",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              null,
              <SubHeaderBarActions
                left={true}
>
                <Action
                                bsStyle="link"
                                className="theme-tc-subheader-back-button tc-subheader-back-button"
                                hideLabel={true}
                                icon="talend-arrow-left"
                                id="backArrow"
                                label="Go Back"
                                onClick={[Function]}
                />
                <InputTitleSubHeader
                                editMode={false}
                                inProgress={false}
                                loading={false}
                                t={[Function]}
                                title="defaultTitle"
                />
</SubHeaderBarActions>,
              false,
              null,
              false,
              null,
              false,
              null,
            ],
            "className": "theme-tc-subheader-navbar tc-subheader-navbar",
          },
          "ref": null,
          "rendered": Array [
            null,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  null,
                  <Action
                    bsStyle="link"
                    className="theme-tc-subheader-back-button tc-subheader-back-button"
                    hideLabel={true}
                    icon="talend-arrow-left"
                    id="backArrow"
                    label="Go Back"
                    onClick={[Function]}
/>,
                  null,
                  <InputTitleSubHeader
                    editMode={false}
                    inProgress={false}
                    loading={false}
                    t={[Function]}
                    title="defaultTitle"
/>,
                  null,
                ],
                "left": true,
              },
              "ref": null,
              "rendered": Array [
                null,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "bsStyle": "link",
                    "className": "theme-tc-subheader-back-button tc-subheader-back-button",
                    "hideLabel": true,
                    "icon": "talend-arrow-left",
                    "id": "backArrow",
                    "label": "Go Back",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                null,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "editMode": false,
                    "inProgress": false,
                    "loading": false,
                    "t": [Function],
                    "title": "defaultTitle",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                null,
              ],
              "type": [Function],
            },
            false,
            null,
            false,
            null,
            false,
            null,
          ],
          "type": [Function],
        },
        null,
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`Layout should render layout with TabBar component 1`] = `
<div
  className="tc-layout theme-layout"
  id={undefined}
>
  <header
    className="tc-layout-header theme-header"
    role="banner"
  >
    <h1>
      Header
    </h1>
  </header>
  <div
    className="tc-layout-two-columns theme-container"
  >
    <div
      className="tc-layout-two-columns-left theme-sidemenu"
    >
      <h1>
        Column one
      </h1>
    </div>
    <div
      className="tc-layout-two-columns-main theme-main"
    >
      <div
        className="theme-tc-with-drawer"
      >
        <nav
          className="nav tc-tab-bar theme-tc-tab-bar"
        >
          <ul
            className="nav nav-tabs tc-tab-bar-actions"
          >
            <li
              className="tc-tab-bar-action"
            >
              <button
                className="btn btn-link"
                disabled={false}
                id="tab-bar-action-1"
                onClick={[Function]}
                onMouseDown={[Function]}
                role={null}
                type="button"
              >
                <span>
                  Tab1
                </span>
              </button>
            </li>
            <li
              className="tc-tab-bar-action active"
            >
              <button
                className="btn btn-link"
                disabled={false}
                id="tab-bar-action-2"
                onClick={[Function]}
                onMouseDown={[Function]}
                role={null}
                type="button"
              >
                <span>
                  Tab2
                </span>
              </button>
            </li>
            <li
              className="tc-tab-bar-action"
            >
              <button
                className="btn btn-link"
                disabled={false}
                id="tab-bar-action-3"
                onClick={[Function]}
                onMouseDown={[Function]}
                role={null}
                type="button"
              >
                <span>
                  Tab3
                </span>
              </button>
            </li>
          </ul>
        </nav>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "height": "100%",
              "overflow": "auto",
            }
          }
        >
          <h1>
            Column two
          </h1>
        </div>
        <div
          className="theme-tc-with-drawer-container"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout should render layout with footer component 1`] = `
<div
  className="tc-layout theme-layout"
  id={undefined}
>
  <header
    className="tc-layout-header theme-header"
    role="banner"
  >
    <h1>
      Header
    </h1>
  </header>
  <div
    className="tc-layout-one-column theme-main"
  >
    <div
      className="theme-tc-with-drawer"
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "overflow": "auto",
          }
        }
      >
        <h1>
          Column one
        </h1>
      </div>
      <div
        className="theme-tc-with-drawer-container"
      />
    </div>
  </div>
  <footer
    className="tc-layout-footer theme-footer"
    role="contentinfo"
  >
    <h3>
      Footer
    </h3>
  </footer>
</div>
`;

exports[`Layout should render layout without header 1`] = `
<div
  className="tc-layout theme-layout"
  id={undefined}
>
  <div
    className="tc-layout-one-column theme-main"
  >
    <div
      className="theme-tc-with-drawer"
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "height": "100%",
            "overflow": "auto",
          }
        }
      >
        <h1>
          Column one
        </h1>
      </div>
      <div
        className="theme-tc-with-drawer-container"
      />
    </div>
  </div>
</div>
`;
